src/pybind/mgr/rook/module.py

```diff
@@ -331,6 +332,36 @@ def get_inventory(self, node_filter=None):
 
         return result
 
+    @deferred_read
+    def list_services(self, service_type):
+
+        assert service_type in ('mds', 'osd', 'mon', 'mgr', None), service_type + ' unsupported'
+
+        pods = self.rook_cluster.list_pods(service_type)
+
+        result = []
+        for p in pods:
+            sl = orchestrator.ServiceLocation()
+            sl.nodename = p['nodename']
+            sl.container_id = p['name']
+            sl.service_type = re.sub('rook-ceph-', '', p['labels']['app'])
+
+            if sl.service_type == 'osd':
+                sl.daemon_name = p['labels']['ceph-osd-id']
+            elif sl.service_type == 'mds':
+                sl.daemon_name = p['labels']["rook_file_system"]
+            elif sl.service_type == 'mon':
+                sl.daemon_name = p['labels']['mon']
+            elif sl.service_type == 'mgr':
+                sl.daemon_name = p['labels']['mgr']
+            else:
```

Comment:
It looks like we should be able to handle rgw here too, but I don't seem to be able to get it to start with the current rook code. Once it's clearer what sort of labels that has, we should be able to add support for rgw as well.